File: datasets/IR-Plag/original-java-rest-kotlin/case-06/plagiarized/L5/06/TestSatuArray.kt
package IR

import java.util.*

object TestSatuArray {
    @JvmStatic
    fun main(args: Array<String>) {
        val sc = Scanner(System.`in`)
        var arr = IntArray(10) //inisialisasi array
        arr = inputArr(arr)
        printArr(arr)
    }

    fun inputArr(arr: IntArray): IntArray {
        val sc = Scanner(System.`in`)
        var x = 0
        while (x < 10) { //looping untuk input array
            print("Read a number: ")
            arr[x] = sc.nextInt() //input ke array
            x++
        }
        return arr
    }

    fun printArr(arr: IntArray) {
        var x = 9
        while (x >= 0) { //looping untuk menampilkan array
            println(arr[x]) //menampilkan array
            x--
        }
    }
}

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-06/plagiarized/L5/06/TestSatuArray.ts
import * as readline from 'readline';

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function inputArr(arr: number[]): number[] {
  let x = 0;
  while (x < 10) { // looping untuk input array
    rl.question('Read a number: ', (input) => {
      arr[x] = parseInt(input);
      x++;
      if (x === 10) {
        rl.close();
        return arr;
      }
    });
  }
  return arr;
}

function printArr(arr: number[]) {
  let x = 9;
  while (x >= 0) { // looping untuk menampilkan array
    console.log(arr[x]); // menampilkan array
    x--;
  }
}

function main() {
  const arr: number[] = new Array(10);
  inputArr(arr);
  printArr(arr);
}

main();

